### PR TITLE
'Is reporter a developer' as a feature

### LINKED ISSUE #83: 'Is reporter a developer' as a feature
Similar to #30.

This feature is going to be a boolean saying whether the reporter of the bug is a developer. As a first implementation, we can check if there is any commit where they are authors.

### PR DESCRIPTION
**Pull request by poojan124**
'Is reporter a developer' as a feature

Fixes #83.

### CODE REVIEW

File: bugbug/bug_features.py [hunk 1]
```diff
@@ -267,6 +267,11 @@ def __call__(self, bug, **kwargs):
         return len(set(component for commit in bug['commits'] for component in commit['components'] if not commit['ever_backedout']))
 
 
+class is_developer(object):
```
Comment (marco-c) [inline bugbug/bug_features.py]: `is_reporter_a_developer`

**Comment by poojan124** [comment]
@marco-c how can i update database here? so that it passes the CI checks.

File: bugbug/bug_features.py (hunk 1 above)
Comment (poojan124) [reply bugbug/bug_features.py]: Done

File: bugbug/bug_features.py [hunk 2]
```diff
@@ -333,6 +345,7 @@ def transform(self, bugs):
         results = []
 
         reporter_experience_map = defaultdict(int)
+        author_ids = get_author_ids()
```
Comment (marco-c) [inline bugbug/bug_features.py]: Only call get_author_ids if commit_data is not None (like we do for `self.commit_map`)

Comment (marco-c) [reply bugbug/bug_features.py]: (this will probably fix CI too)

File: bugbug/repository.py [hunk 3]
```diff
@@ -55,6 +55,7 @@ def _transform(commit):
         'components': list(),
         'author_experience': author_experience[commit],
         'author_experience_90_days': author_experience_90_days[commit],
+        'author_email': commit.author_email.decond('utf-8'),
```
Comment (marco-c) [inline bugbug/repository.py]: Did you try to run the script? `decond` must be wrong :P

File: bugbug/bug_features.py (hunk 2 above)
Comment (poojan124) [reply bugbug/bug_features.py]: i dont get it which condition to put here? error saying no author_email in commits.

Comment (marco-c) [reply bugbug/bug_features.py]: the same we have a few lines earlier for `self.commit_map`.

Comment (poojan124) [reply bugbug/bug_features.py]: Done

File: bugbug/repository.py (hunk 3 above)
Comment (poojan124) [reply bugbug/repository.py]: ;_; Done

File: bugbug/bug_features.py [hunk 4]
```diff
@@ -333,6 +345,7 @@ def transform(self, bugs):
         results = []
 
         reporter_experience_map = defaultdict(int)
+        author_ids = None if self.commit_map is None else get_author_ids()
```
Comment (marco-c) [inline bugbug/bug_features.py]: Do it the other way around, like for commit_map, it is more readable

```Python
author_ids = get_author_ids() if self.commit_map else None
```

Comment (poojan124) [reply bugbug/bug_features.py]: This is not working if commit_map is empty.
```
cm = {}
>>> 1 if cm else 0
0
```
I can do this instead
```
>>> 1 if cm is not None else 0
```

Comment (marco-c) [reply bugbug/bug_features.py]: When is commit_map empty?

Comment (poojan124) [reply bugbug/bug_features.py]: Ok so i mined small data(1000) using new template, and all those commits contains `bug_id = ""` so `get_commit_map` is returning `{} dictionary. `     
So you want to me to continue with this one? 
`author_ids = get_author_ids() if self.commit_map else None`

Comment (marco-c) [reply bugbug/bug_features.py]: Ah ok, yeah in general it should never be empty, unless you have wrong data in the commits.json DB (you might want to redownload it).

**Comment by poojan124** [comment]
@marco-c I think all good now.

**Pull request merged**